tesseract/tesseract_urdf/include/tesseract_urdf/safety_controller.h

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file safety_controller.h
+ * @brief Parse safety_controller from xml string
+ *
+ * @author Levi Armstrong
+ * @date Dec 18, 2017
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2017, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TESSERACT_URDF_SAFETY_CONTROLLER_H
+#define TESSERACT_URDF_SAFETY_CONTROLLER_H
+
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <tesseract_common/status_code.h>
+#include <Eigen/Geometry>
+#include <tinyxml2.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_scene_graph/joint.h>
+
+namespace tesseract_urdf
+{
+class SafetyStatusCategory : public tesseract_common::StatusCategory
+{
+public:
+  SafetyStatusCategory() : name_("SafetyStatusCategory") {}
+  const std::string& name() const noexcept override { return name_; }
+  std::string message(int code) const override
+  {
+    switch (code)
+    {
+      case Success:
+        return "Sucessful";
+      case MissingAttributeAllOptional:
+        return "Missing safety attribute 'soft_upper_limit', 'soft_lower_limit', and 'k_position', using default value "
+               "0, 0, and 1!";
+      case MissingAttributeSoftUpperLimit:
+        return "Missing safety attribute 'soft_upper_limit', using default value 0!";
+      case MissingAttributeSoftLowerLimit:
+        return "Missing safety attribute 'soft_lower_limit', using default value 0!";
+      case MissingAttributeKPosition:
+        return "Missing safety attribute 'k_position', using default value 0!";
+      case ErrorAttributeKVelocity:
+        return "Missing or failed to parse safety attribute 'k_velocity'!";
+      default:
+        return "Invalid error code for " + name_ + "!";
+    }
+  }
+
+  enum
+  {
+    MissingAttributeAllOptional = 4,
+    MissingAttributeSoftUpperLimit = 3,
+    MissingAttributeSoftLowerLimit = 2,
+    MissingAttributeKPosition = 1,
```

Comment:
If missing 2 of the 3 only the first gets the status code. I don't see that as a huge deal, but I wanted to comment on it.

Reply:
update provide all missing status codes.